src/main/java/ch/njol/skript/expressions/ExprTotalExperience.java

```diff
@@ -0,0 +1,130 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.bukkitutil.PlayerUtils;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.ExperienceOrb;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Total Experience")
+@Description({
+	"The total experience, in points, of players or experience orbs.",
+	"Adding to a player's experience will trigger Mending, but setting their experience will not."
+})
+@Examples({
+	"set total experience of player to 100",
+	"",
+	"add 100 to player's experience",
+	"",
+	"if player's total experience is greater than 100:",
+	"\tset player's total experience to 0",
+	"\tgive player 1 diamond"
+})
+@Since("INSERT VERSION")
+public class ExprTotalExperience extends SimplePropertyExpression<Entity, Integer> {
+
+	static {
+		register(ExprTotalExperience.class, Integer.class, "[total] experience", "entities");
+	}
+
+	@Override
+	@Nullable
+	public Integer convert(Entity entity) {
+		// experience orbs
+		if (entity instanceof ExperienceOrb)
+			return ((ExperienceOrb) entity).getExperience();
+
+		// players need special treatment
+		if (entity instanceof Player)
+			return PlayerUtils.getTotalXP(((Player) entity).getLevel(), ((Player) entity).getExp());
+
+		// invalid entity type
+		return null;
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (mode == ChangeMode.SET || mode == ChangeMode.ADD || mode == ChangeMode.REMOVE || mode == ChangeMode.RESET || mode == ChangeMode.DELETE)
```

Comment:
Why don't you make this negated for REMOVE_ALL

Reply:
unless I miss understanding I believe this falls in line with this comment https://github.com/SkriptLang/Skript/pull/5513#discussion_r1133405621

Reply:
Then a switch should be in place